### PR TITLE
[EFR32] First draft of a functioning light switch example

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* No light switch examples were available for efr32

#### Change overview
* Add a light switch example that can send a toggle command to a single bound lighting app

#### Testing
* Manual tests with two efr32

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Approved and fast tracking a platform change but a platform maintainer. Note, still fails restyle.

File: examples/light-switch-app/efr32/args.gni [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+import("${chip_root}/src/platform/EFR32/args.gni")
+
+efr32_sdk_target = get_label_info(":sdk", "label_no_toolchain")
+
+pw_log_BACKEND = "${chip_root}/src/lib/support/pw_log_chip"
+pw_assert_BACKEND = "$dir_pw_assert_log"
+chip_enable_openthread = true
+chip_openthread_ftd = true
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/efr32/args.gni:24]: to allow configuration of FTTD or MTD and SED chip_openthread_ftd is moved to the .gn file. It should be the same here

File: examples/light-switch-app/efr32/include/CHIPProjectConfig.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+ * CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID
+ *
+ * 0x235A: Chip's Vendor Id.
+ */
+#define CHIP_DEVICE_CONFIG_DEVICE_VENDOR_ID 0x235A
+
+/**
+ * CHIP_DEVICE_CONFIG_DEVICE_PRODUCT_ID
+ *
+ * 0x534C: EFR32 lighting-app
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/efr32/include/CHIPProjectConfig.h:59]: Should use a new code

File: examples/light-switch-app/efr32/include/FreeRTOSConfig.h [hunk 3]
```diff
@@ -0,0 +1,303 @@
+/*
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/efr32/include/FreeRTOSConfig.h:1]: This file is outdated for power_management/SED build

File: examples/light-switch-app/efr32/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+#if CHIP_ENABLE_OPENTHREAD
+    EFR32_LOG("Initializing OpenThread stack");
+    ret = ThreadStackMgr().InitThreadStack();
+    if (ret != CHIP_NO_ERROR)
+    {
+        EFR32_LOG("ThreadStackMgr().InitThreadStack() failed");
+        appError(ret);
+    }
+
+    ret = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_Router);
```
Comment (jmartinez-silabs) [inline examples/light-switch-app/efr32/src/main.cpp:153]: FTD/MTD/SED type selection to be done here

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]
Looks good, just need the latest changes for SED and power management support

File: examples/light-switch-app/efr32/args.gni (hunk 1 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/efr32/args.gni:24]: done

File: examples/light-switch-app/efr32/include/FreeRTOSConfig.h (hunk 3 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/efr32/include/FreeRTOSConfig.h:1]: done

File: examples/light-switch-app/efr32/src/main.cpp (hunk 4 above)
Comment (mkardous-silabs) [reply examples/light-switch-app/efr32/src/main.cpp:153]: done

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**